test/system/src/main/java/io/pravega/test/system/framework/DockerRemoteSequential.java

```diff
@@ -0,0 +1,154 @@
+    public final static DockerClient CLIENT = DefaultDockerClient.builder().uri("http://"+ getConfig("masterIP", "Invalid Master IP")+ ":2375").build();
+    private final static String IMAGE = "java:8";
+    public String id;
+    private final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(3);
+
+    public CompletableFuture<Void> startTestExecution(Method testMethod) {
+
+        log.debug("Starting test execution for method: {}", testMethod);
+
+        String className = testMethod.getDeclaringClass().getName();
+        String methodName = testMethod.getName();
+        String containerName = methodName + ".testJob";
+
+        return CompletableFuture.runAsync(() -> {
+            startTest(containerName, className, methodName);
+        }).thenCompose(v2 -> waitForJobCompletion(containerName))
+                .whenComplete((v, ex) -> {
+            stopContainer(); //stop container once execution is complete.
+                    if (ex != null) {
+                        log.error("Error while executing the test. ClassName: {}, MethodName: {}", className,
+                                methodName);
+                    }
+                }
+                );
+    }
+
+
+    @Override
+    public CompletableFuture<Void> stopTestExecution(String testID) {
+        throw new NotImplementedException("Stop Execution is not used for Remote sequential execution");
+    }
+
+    private CompletableFuture<Void> waitForJobCompletion(final String containerName) {
+        return FutureHelpers.loop(() -> isTestRunning(containerName),
+                () -> FutureHelpers.delayedFuture(Duration.ofSeconds(3), executorService),
+                executorService);
+    }
+
+    private boolean isTestRunning(String containerName) {
+        boolean value = false;
+        try {
+            ContainerInfo info = CLIENT.inspectContainer(id);
+                if (info.name().equals(containerName)) {
+                    value = true;
+                }
+        } catch (DockerException | InterruptedException e) {
+            log.error("unable to list docker services ", e);
+        }
+        return value;
+    }
+
+    private String startTest(String containerName, String className, String methodName) {
+
+        try {
+            CLIENT.pull(IMAGE);
+
+            ContainerCreation containerCreation = CLIENT.createContainer(setContainerConfig(methodName, className), containerName);
+            assertFalse(containerCreation.id().toString().equals(null));
+
+              id = containerCreation.id();
+
+            final Path dockerDirectory = Paths.get(System.getProperty("user.dir"));
+            try {
+                CLIENT.copyToContainer(dockerDirectory, id, "/data");
+            } catch (IOException | DockerException | InterruptedException e) {
+                log.error("Exception while copying test jar to the container ", e);
+                Assert.fail("Unable to copy test jar to the container.Test failure");
```

Comment:
Here an elsewhere when an exception causes a test to fail, it is better to do something like `new AssertionError("...",e);` or similar because it allows the stack trace to show up when someone sees the failure in Gradle, or Jankens, or their IDE.

Reply:
done.